registration/include/pcl/registration/impl/ia_fpcs.hpp

```diff
@@ -451,24 +451,20 @@ pcl::registration::FPCSInitialAlignment<PointSource, PointTarget, NormalT, Scala
     setupBase(pcl::Indices& base_indices, float (&ratio)[2])
 {
   float best_t = FLT_MAX;
-  const std::vector<int> copy(base_indices.begin(), base_indices.end());
+  const pcl::Indices copy(base_indices.begin(), base_indices.end());
   pcl::Indices temp(base_indices.begin(), base_indices.end());
 
   // loop over all combinations of base points
-  for (std::vector<int>::const_iterator i = copy.begin(), i_e = copy.end(); i != i_e;
-       ++i)
-    for (std::vector<int>::const_iterator j = copy.begin(), j_e = copy.end(); j != j_e;
-         ++j) {
+  for (auto i = copy.begin(), i_e = copy.end(); i != i_e; ++i)
```

Comment:
If indices in `copy` are guaranteed to be unique (implicit assumption in pcl), we can replace all iterators with references and a range-for loop

Reply:
Maybe, but `i`, `j`, and `k` are also compared with the iterator `l` further down. I think I would rather not make any further changes here since the code is a bit complicated and I don't want to introduce bugs. Maybe someone can look over it again in the future.

Reply:
Yeah. Forgot to mention: not for this PR